presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -97,7 +123,7 @@ public synchronized boolean tryReserveMemory(long bytes)
     {
         checkArgument(bytes >= 0, "bytes is negative");
 
-        if (reserved + bytes > maxMemory) {
+        if (willExceedMemoryLimit(bytes)) {
```

Comment:
it uses different condition than in: https://github.com/prestodb/presto/pull/5711/commits/2a4d2197501b59e7c74ff3b863ff88483de7ab4e#diff-68ac3b293142f2c6f9d1f477192b9634L79